engine/proofview.ml

```diff
@@ -1280,7 +1280,7 @@ module V82 = struct
    { Evd.it = List.map drop_state comb ; sigma = solution }
 
   let top_goals initial { solution=solution; } =
-    let goals = CList.map (fun (t,_) -> fst (Constr.destEvar (EConstr.Unsafe.to_constr t))) initial in
+    let goals = CList.map (fun (t,_) -> fst (EConstr.destEvar solution t)) initial in
```

Comment:
Just revert this change, this function is unsafe and can be looked at in another PR.

Reply:
Done